src/System.Memory/src/System/SpanDebugView.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace System
+{
+    internal sealed class SpanDebugView<T>
+    {
+        private readonly T[] _pinnable;
+        private readonly IntPtr _byteOffset;
+        private readonly int _length;
+
+        public SpanDebugView(Span<T> collection)
+        {
+            _pinnable = (T[])(object)collection.Pinnable;
+            _byteOffset = collection.ByteOffset;
+            _length = collection.Length;
+        }
+
+        [DebuggerBrowsable(DebuggerBrowsableState.RootHidden)]
+        public unsafe T[] Items
+        {
+            get
+            {
+                int elementSize = typeof(T).IsValueType ? Unsafe.SizeOf<T>() : IntPtr.Size;//Workaround to VIL bug where Unsafe.SizeOf<T> where T is a class, returns the size of its fields instead of IntPtr.Size
+
+                if (_pinnable == null)
+                {
+                    var result = new T[_length];
+                    byte* source = (byte*)_byteOffset.ToPointer();
+                    byte* fixedResult = (byte*)Unsafe.AsPointer(ref result[0]);
+
+                    int totalSize = _length * elementSize;
+
+                    for (int i = 0; i < totalSize; i++)
+                    {
+                        *fixedResult = *source;
```

Comment:
This will corrupt GC tracking if T contains object references. You have to copy the elements as `T`, not as bytes.